sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -249,16 +330,74 @@
   }
 
   /**
+   * Creates an uninitialized {@link Read} {@link PTransform}, using a Kafka {@link Deserializer}
+   * based on {@link AvroCoder}. The coder reads Avro data directly without using an Avro object
```

Comment:
It's probably worth mentioning what guarantees on the data wire format this gives and doesn't give. Is this intended to read only data produced by KafkaIO.toAvro()?